android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt

```diff
@@ -251,11 +279,112 @@ class QuestionnaireActivity : BaseMultiLanguageActivity() {
             )
             finish()
           }
+          // do background location processing
+          // check here that "logQuestionnaireLocation": true
         }
       }
     }
   }
 
+  private fun getLocation() {
```

Comment:
Is it possible to extract this functionality for requesting permissions to a utility function for re-use in other places that need explicit permissions?

I'm okay with refining this implementation in the next phase.